### PR TITLE
Retire timestep

### PR DESCRIPTION
**Pull request by KineticTheory**
### Background

+ I don't think any Draco client software packages use timestep, nor have they
  used this package for the last dozen years or so.  It it time to remove it.

### Status

* Reference:
  * [Pre-Merge Code Review](https://github.com/lanl/Draco/wiki/Style-Guide)
  * [CDash Status](https://rtt.lanl.gov/cdash/index.php?project=Draco&filtercount=1&showfilters=1&field1=buildname&compare1=63&value1=-pr)
* Checks
  * [ ] Travis/Appveyor CI checks pass
  * [ ] Code coverage does not decrease
  * [ ] [Valgrind test passes](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [ ] [Toss3 checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [ ] [Trinitite checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [ ] Code reviewed/approved, sufficient DbC checks, testing, documentation

### CODE REVIEW

File: config/component_macros.cmake [hunk 1]
```diff
@@ -638,199 +633,6 @@ macro( register_parallel_test targetname numPE command cmd_args )
   unset( lverbose )
 endmacro()
 
-#----------------------------------------------------------------------#
```
Comment (keadyk) [inline config/component_macros.cmake]: This seems unrelated to the timestep retirement -- I just want to be sure that it's an intentional change and not a diff from somewhere else!

Comment (KineticTheory) [reply config/component_macros.cmake]: This was intentional, but now I'm having 2nd thoughts about it.  Marking this PR as WIP.  There is no urgency with this one.  

As a reminder, I will probably start the next release cycle on Monday.

Comment (KineticTheory) [reply config/component_macros.cmake]: Okay - I fixed this PR so that it only has _timestep_ specific deletions.

**Comment by keadyk** [comment]
LGTM! :)
